File: backend/app/schemas/audit.py
import uuid
from datetime import datetime
from decimal import Decimal
from typing import Optional
from pydantic import BaseModel, ConfigDict
from app.core.enums import AuditCategory, AuditSeverity, EventProvenance


class AuditLogResponse(BaseModel):
    id: uuid.UUID
    trace_id: str
    merchant_id: uuid.UUID
    customer_id: Optional[uuid.UUID] = None
    session_id: Optional[uuid.UUID] = None
    order_id: Optional[uuid.UUID] = None
    agent_source: str
    event_category: AuditCategory
    event_type: str
    summary: str
    financial_delta: Decimal
    provenance: EventProvenance
    severity: AuditSeverity
    metadata_json: dict
    timestamp: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/audit.ts
import { EventProvenance } from "./commerce";

export type AuditCategory = 
  | "GROWTH_ACTION"
  | "RECOVERY_ACTION"
  | "POLICY_CHECK"
  | "PAYMENT_EVENT"
  | "MERCHANT_APPROVAL"
  | "SYSTEM_EVENT";

export type AuditSeverity = "INFO" | "WARNING" | "CRITICAL";

export interface AuditLog {
  id: string;
  trace_id: string;
  merchant_id: string;
  customer_id?: string;
  session_id?: string;
  order_id?: string;
  agent_source: string;
  event_category: AuditCategory;
  event_type: string;
  summary: string;
  financial_delta: number;
  provenance: EventProvenance;
  severity: AuditSeverity;
  metadata_json: Record<string, any>;
  timestamp: string;
}
